src/resmom_win/start_exec.c

```diff
@@ -3131,2351 +3142,201 @@ open_std_file(job *pjob, enum job_file which, int mode, gid_t exgid)
-			strcpy(path, jobdirname(pjob->ji_qs.ji_jobid, pjob->ji_grpcache->gc_homedir));
-		} else {        /* force all output to user's HOME */
-			(void)strcpy(path, pjob->ji_grpcache->gc_homedir);
-		}
-		(void)strcat(path, "/");
-		*keeping = 1;
-#else	/* NO_SPOOL_OUTPUT */
-		/* sandbox=PRIVATE mode puts output in job staging and execution directory */
-		if ((pjob->ji_wattr[(int)JOB_ATR_sandbox].at_flags&ATR_VFLAG_SET) &&
-			(strcasecmp(pjob->ji_wattr[(int)JOB_ATR_sandbox].at_val.at_str, "PRIVATE") == 0)) {
-			strcpy(path, jobdirname(pjob->ji_qs.ji_jobid, pjob->ji_grpcache->gc_homedir));
-			strcat(path, "/");
-		} else {
-			(void)strcpy(path, path_spool);
-		}
-		*keeping = 0;
-#endif	/* NO_SPOOL_OUTPUT */
-		if (*pjob->ji_qs.ji_fileprefix != '\0')
-			(void)strcat(path, pjob->ji_qs.ji_fileprefix);
-		else
-			(void)strcat(path, pjob->ji_qs.ji_jobid);
-		(void)strcat(path, suffix);
-	}
-	return (path);
-}
-
-/**
- * @brief
- *      Open (create) either standard output or standard error for the job.
- *
- * @param[in] pjob  - pointer to job structure
- * @param[in] which - which file to create, StdOut, StdErr, or Chkpt
- * @param[in] mode  - file open oflag (O_CREAT, O_WRONLY, ...)
- * @param[in] exgid - User's gid
- *
- * @return      int
- * @retval      fd      On success
- * @retval      -1      on failure
- *
- */
-int
-open_std_file(job *pjob, enum job_file which, int mode, gid_t exgid)
-{
-	int   fds;
-	int   keeping;
-	char *path;
-
-	path = std_file_name(pjob, which, &keeping);
-	if (keeping) {
-		/* in user's home,  may be NFS mounted, must create as user */
-
-#if defined(HAVE_SETEUID) && defined(HAVE_SETEGID)
-		/* most systems */
-		if ((setegid(exgid) == -1) ||
-			(seteuid(pjob->ji_grpcache->gc_uid) == -1))
-			return -1;
-		fds = open(path, mode, 0666);
-		(void)seteuid(0);
-		(void)setegid(pbsgroup);
-#elif defined(HAVE_SETRESUID) && defined(HAVE_SETRESGID)
-		/* HPUX and the like */
-		if ((setresgid(-1, exgid, -1) == -1) ||
-			(setresuid(-1, pjob->ji_grpcache->gc_uid, -1) == -1))
-			return -1;
-		fds = open(path, mode, 0666);
-		(void)setresuid(-1, 0, -1);
-		(void)setresgid(-1, pbsgroup, -1);
-#else	/* Neither */
-		Crash and Burn - need seteuid/setegid   or need setresuid/setresgid
-#endif	/* HAVE_SETRESUID */
-	} else {
-		fds = open(path, mode, 0666);
-		if (fds >= 0) {
-			/* change file uid/gid to execution user of job  */
-			(void)fchown(fds, pjob->ji_qs.ji_un.ji_momt.ji_exuid, exgid);
-		}
-	}
-	return (fds);
-}
-
-/**
- * @brief
- *      find_env_slot - find if the environment variable is already in the table,
- *      If so, replace the existing one with the new one.
- *
- * @param[in] ptbl - pointer to var_table which holds environment variable for job
- * @param[in] pstr - new environment variable
- *
- * @return      int
- * @retval      !(-1)   success
- * @retval      -1      Failure
- *
- */
-
-static int
-find_env_slot(struct var_table *ptbl, char *pstr)
-{
-	int	 i;
-	int	 len = 1;	/* one extra for '=' 
```

Comment:
can you explain why some of these functions have moved around? it is really hard to review the code with such a huge movement. Little explanation would be helpful.